### PR TITLE
dnsdist: Ensure we have at least one protobuf MetaValue

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

The `MetaValue` is required to exist, which means that we need to have at least one value.

The DoH-related values should be empty non-DoH protocols, but the protocol buffer messages should still be sent, with the expected content.
Fixes https://github.com/PowerDNS/pdns/issues/12560

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Rebased on master to fix CI issues.

**Review by fredmorcos — APPROVED** [review]
Looks OK to me. But I'd prefer waiting for a second review.

**Review by omoerbeek — APPROVED** [review]

**Pull request merged**
